service/service.go

```diff
@@ -0,0 +1,190 @@
+		return errors.Wrap(err, "unable to register checkers")
+	}
+
+	svc.router = mux.NewRouter()
+	//svc.server = svc.serviceList.GetHTTPServer(svc.config.BindAddr, svc.router)
+	svc.router.StrictSlash(true).Path("/health").HandlerFunc(svc.healthCheck.Handler)
+	svc.healthCheck.Start(ctx)
+
+	//Initialise transformer
+	transformer := transformer.New()
+
+	//Initialse elasticSearchClient
+	elasticSearchClient := elasticsearch.New(svc.config.ElasticSearchAPIURL, dphttp.NewClient(), svc.config.SignElasticsearchRequests, svc.esSigner, svc.config.AwsRegion, svc.config.AwsService)
+
+	//Initialse AWS signer
+	if svc.config.SignElasticsearchRequests {
+		svc.esSigner, err = esauth.NewAwsSigner("", "", svc.config.AwsRegion, svc.config.AwsService)
+		if err != nil {
+			log.Error(ctx, "failed to create aws v4 signer", err)
+			os.Exit(1)
+		}
+	}
+
+	//Initialise query builder
+	queryBuilder, err := query.NewQueryBuilder()
+	if err != nil {
+		log.Fatal(ctx, "error initialising query builder", err)
+		os.Exit(1)
+	}
+
+	//Create Search API
+	if err := api.CreateAndInitialise(svc.config, svc.router, queryBuilder, elasticSearchClient, transformer, svcErrors); err != nil {
+		log.Fatal(ctx, "error initialising API", err)
+		os.Exit(1)
+	}
+
+	return nil
+}
+
+// Close gracefully shuts the service down in the required order, with timeout
+func (svc *Service) Close(ctx context.Context) error {
+	timeout := svc.config.GracefulShutdownTimeout
+	log.Info(ctx, "commencing graceful shutdown", log.Data{"graceful_shutdown_timeout": timeout})
+	shutdownContext, cancel := context.WithTimeout(ctx, timeout)
+	hasShutdownError := false
+
+	// Gracefully shutdown the application closing any open resources.
+	go func() {
+		defer cancel()
+
+		// stop healthcheck, as it depends on everything else
+		if svc.serviceList.HealthCheck {
+			svc.healthCheck.Stop()
+		}
+
+		// stop any incoming requests before closing any outbound connections
+		//if err := svc.server.Shutdown(shutdownContext); err != nil {
+		//	log.Error(shutdownContext, "error closing API", err)
+		//	hasShutdownError = true
+		//}
+
+	}()
+
+	// wait for shutdown success (via cancel) or failure (timeout)
+	<-shutdownContext.Done()
+
+	// timeout expired
+	if shutdownContext.Err() == context.DeadlineExceeded {
+		log.Error(shutdownContext, "shutdown timed out", shutdownContext.Err())
+		return shutdownContext.Err()
+	}
+
+	// other error
+	if hasShutdownError {
+		err := errors.New("failed to shutdown gracefully")
+		log.Error(shutdownContext, "failed to shutdown gracefully ", err)
+		return err
+	}
+
+	log.Info(shutdownContext, "graceful shutdown was successful")
+	cancel()
+	return nil
+}
+
+func (svc *Service) registerCheckers(ctx context.Context, elasticHTTPClient dphttp.Clienter) (err error) {
+	//cfg *config.Configuration,
+	//elasticHTTPClient dphttp.Clienter,
+	//esSigner *esauth.Signer,
+	//svcList *service.ExternalServiceList)
+
+	hasErrors := false
+	elasticHTTPClient = dphttp.NewClient()
```

Comment:
We shouldn't be creating a new client, we should use the passed in elasticHTTPClient.

The client used for checking if elasticsearch is healthy, should be the same client used for querying elasticsearch otherwise if 1 client was wrong in anyway this wouldn't necessarily reflect the other client resulting in healthcheck failing yet search queries are working or worse the other way round, healthcheck is passing but search queries are failing (due to client).